component/azstorage/block_blob.go

```diff
@@ -654,7 +655,7 @@ func (bb *BlockBlob) getBlobAttr(blobInfo *container.BlobItem) (*internal.ObjAtt
 		log.Trace("BlockBlob::List : blob is encrypted with customer provided key so fetching metadata explicitly using REST")
 		return bb.getAttrUsingRest(*blobInfo.Name)
 	}
-	mode, err := bb.getFileMode(blobInfo.Properties.Permissions)
+	mode, err := bb.getFileMode(blobInfo.Properties.Permissions, blobInfo.Metadata)
```

Comment:
The `permissions` metadata parsing is missing in the `getAttrUsingRest` method. Please check